src/main/java/net/fabricmc/loom/configuration/providers/forge/MinecraftPatchedProvider.java

```diff
@@ -444,35 +486,32 @@ public String side() {
 		}
 	}
 
-	private void remapPatchedJar(Logger logger) throws Exception {
-		Path[] libraries = MinecraftMappedProvider.getRemapClasspath(getProject());
-		logger.lifecycle(":remapping minecraft (TinyRemapper, srg -> official)");
-		TinyTree mappingsWithSrg = getExtension().getMappingsProvider().getMappingsWithSrg();
-
-		Path input = minecraftMergedPatchedSrgAtJar.toPath();
-		Path output = minecraftMergedPatchedJar.toPath();
-
-		Files.deleteIfExists(output);
-
-		TinyRemapper remapper = TinyRemapper.newRemapper()
-				.logger(getProject().getLogger()::lifecycle)
-				.logUnknownInvokeDynamic(false)
-					.withMappings(TinyRemapperMappingsHelper.create(mappingsWithSrg, "srg", "official", true))
-				.withMappings(InnerClassRemapper.of(input, mappingsWithSrg, "srg", "official"))
-				.renameInvalidLocals(true)
-				.rebuildSourceFilenames(true)
-				.fixPackageAccess(true)
-				.build();
-
-		try (OutputConsumerPath outputConsumer = new OutputConsumerPath.Builder(output).build()) {
+	private void remapPatchedJar(Path input, Logger logger) throws Exception {
+		Path mcOutput = minecraftMergedPatchedJar.toPath();
+		Path forgeOutput = forgeMergedJar.toPath();
+		Path forgeJar = getForgeJar().toPath();
+		Path forgeUserdevJar = getForgeUserdevJar().toPath();
+		Files.deleteIfExists(mcOutput);
+		Files.deleteIfExists(forgeOutput);
+		TinyRemapper remapper = buildRemapper(input);
+
+		try (OutputConsumerPath outputConsumer = new OutputConsumerPath.Builder(mcOutput).build();
+					OutputConsumerPath outputConsumerForge = new OutputConsumerPath.Builder(forgeOutput).build()) {
 			outputConsumer.addNonClassFiles(input);
 
-			remapper.readClassPath(libraries);
-			remapper.readInputs(input);
-			remapper.apply(outputConsumer);
+			InputTag mcTag = remapper.createInputTag();
+			InputTag forgeTag = remapper.createInputTag();
+			remapper.readInputs(mcTag, input);
+			remapper.readInputs(forgeTag, forgeJar, forgeUserdevJar);
```

Comment:
Possibly readInputsAsync here? (Not sure how it affects the perf with only two calls though)